### PR TITLE
Updated Readme

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Updated Readme.

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/18439

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,46 +1,72 @@
...
-- lint
-  - Install lint on the local environment:
-```bash
-curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b $GOPATH/bin
-```    
+- [Go](https://go.dev/)
+- [Docker](https://www.docker.com/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/)
+- [kustomize](https://kustomize.io/)
+- Access to Kubernetes cluster ([k3d](https://k3d.io/) / k8s)
```
Comment (friedrichwilken) [inline README.md:13]:
```suggestion
- Access to a Kubernetes cluster (e.g. [k3d](https://k3d.io/) / k8s)  
```

File: README.md [hunk 2]
```diff
@@ -1,46 +1,72 @@
...
 
 ### Build
 
 ```bash
-$ go mod vendor
+make build
 ```
 
-### Test
+### Run tests
```
Comment (NHingerl) [inline README.md:23]:
```suggestion
### Run Tests
```
Capitalization guidelines recently changed - we now use title case for headlines.

File: README.md [hunk 3]
```diff
@@ -1,46 +1,72 @@
...
+
+1. Fix common lint issues:
+
+   ```sh
+   make imports
+   make fmt
+   make lint
+   ```
+
+### Build container images
```
Comment (NHingerl) [inline README.md:41]:
```suggestion
### Build Container Images
```

File: README.md [hunk 4]
```diff
@@ -1,46 +1,72 @@
...
+
+Build and push your image to the location specified by `IMG`:
+
+```sh
+make docker-build docker-push IMG=<container-registry>/eventing-publisher-proxy:<tag> # If using docker, <container-registry> is your username.
+```
+
+> **NOTE**: For MacBook M1 devices, run:
```
Comment (NHingerl) [inline README.md:49]:
```suggestion
For MacBook M1 devices, run:
```
Either put the code block into the Note box as well, or simply drop the note box. My suggestion is for the simpler solution :D

File: README.md [hunk 5]
```diff
@@ -1,46 +1,72 @@
...
+
+```sh
+make docker-buildx IMG=<container-registry>/eventing-publisher-proxy:<tag>
 ```
 
+## Deployment
+
+You need a Kubernetes cluster to run against. You can use [k3d](https://k3d.io/) to get a local cluster for testing, or run against a remote cluster.
```
Comment (NHingerl) [inline README.md:57]: Isn't this already listed in the ## Prerequisites section? Why mention it again here?

File: README.md [hunk 6]
```diff
@@ -1,46 +1,72 @@
...
+
 ### Deploy inside a cluster
```
Comment (NHingerl) [inline README.md:59]:
```suggestion
### Deploy Inside a Cluster
```

File: README.md [hunk 7]
```diff
@@ -1,46 +1,72 @@
...
 ```bash
 $ ko apply -f config/event-publisher-proxy/
 ```
 
+## Usage
+
 ### Send Events
 
-This command supports **CloudEvents**: 
+This command supports **CloudEvents**:
```
Comment (NHingerl) [inline README.md:69]:
```suggestion
The following command supports **CloudEvents**:
```

**Review by friedrichwilken — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (friedrichwilken) [reply README.md]: On the other hand, if @NHingerl was okay with it, I am okay with it...

File: README.md (hunk 5 above)
Comment (mfaizanse) [reply README.md:57]: I just followed the pattern from our other repos ([example](https://github.com/kyma-project/eventing-manager#deployment)), I can remove it?

Comment (NHingerl) [reply README.md:57]: Let's keep consistency for now. In the long term, we can look into removing redundancy in all repos.

**Review by NHingerl — APPROVED** [review]

**Pull request merged**